consensus/hotstuff/votecollector/combined_vote_processor_v2.go

```diff
@@ -141,17 +143,28 @@ func (p *CombinedVoteProcessorV2) Status() hotstuff.VoteCollectorStatus {
 // Design of this function is event driven: as soon as we collect enough signatures to create a QC we will immediately do so
 // and submit it via callback for further processing.
 // Expected error returns during normal operations:
-// * VoteForIncompatibleBlockError - submitted vote for incompatible block
-// * VoteForIncompatibleViewError - submitted vote for incompatible view
-// * model.InvalidVoteError - submitted vote with invalid signature
-// * model.DuplicatedSignerError if the signer has been already added
+//   - [VoteForIncompatibleBlockError] - submitted vote for incompatible block
+//   - [VoteForIncompatibleViewError] - submitted vote for incompatible view
+//   - [model.InvalidVoteError] - submitted vote with invalid signature
+//   - [model.DuplicatedSignerError] if the signer has been already added
+//   - [model.DoubleVoteError] - indicates that the voter has equivocated and submitted different votes for the same view.
```

Comment:
I would suggest that the VoteProcessor rejects all votes that are for different blocks, not trying to detect equivocation for blocks other than its own block. Detecting equivocation for different blocks is in my mind mostly the responsibility of the VoteCollector. In contrast, what we are here mostly concerned about are double-counting attacks.